src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -34,30 +48,68 @@ public override Task<Unit> Handle(DidChangeTextDocumentParams request, Cancellat
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, contents);
+            var documentUri = request.TextDocument.Uri;
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.AddOrUpdate(documentUri, (documentUri) => configuration, (documentUri, prevConfiguration) => configuration);
+            }
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidOpenTextDocumentParams request, CancellationToken cancellationToken)
         {
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.TryAdd(documentUri, configuration);
+            }
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidSaveTextDocumentParams request, CancellationToken cancellationToken)
         {
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri) &&
+                activeBicepConfigCache.TryRemove(documentUri, out RootConfiguration? prevBicepConfiguration) &&
+                prevBicepConfiguration != null)
+            {
+                var curConfiguration = configurationManager.GetConfiguration(documentUri.ToUri());
```

Comment:
ditto

Reply:
Added error handling to not block compilation:
https://github.com/Azure/bicep/commit/979995aec3298bb62719b50267e746b3fd667ffb

Reply:
The addition of the try catches everywhere made this code kind of messy. Plus, future refactoring could easily regress the exception swallowing.

I think we can get a better separation of concerns and also solve the problem I just mentioned by creating a new interface that gets called from the BicepTextDocumentSyncHandler. At the entry point methods into the new class, we could have the try catch blocks to swallow the exception just once per entry point. Inside, you would be free to refactor the logic however you saw fit without worrying about regressing the exception swallowing.

Separately from that, the swallowed exceptions should be traced out as well. You can use the static `Trace` class. It's already hooked up. info doesn't show up in VS unless you enable it via an env var.